launch_testing/launch_testing/__init__.py

```diff
@@ -61,13 +63,10 @@ def on_stdout_lines(self, lines):
         if self.regex_match and not self.matched:
             self.matched = re.search(self.expected_output, self.stdout_data.getvalue())
 
-        if self.matched:
-            # We have enough output to compare; shut down my child
+        if self.matched and self.test_process:
+            # We matched and we're in charge; shut it all down
             for td in self.launch_descriptor.task_descriptors:
-                if td.name == self.name:
-                    td.terminate()
-                    self.killed = True
-                    return
+                td.terminate()
```

Comment:
My conclusion would be that we need a different exit handler then which doesn't care about the exit code.